mypy/semanal.py

```diff
@@ -4715,6 +4712,7 @@ def qualified_name(self, name: str) -> str:
         else:
             return self.cur_mod_id + '.' + name
 
+    @contextmanager
     def enter(self, function: Union[FuncItem, GeneratorExpr, DictionaryComprehension]) -> None:
```

Comment:
The return type doesn't look right. I suspect the issue is that self check isn't running right now at all, so these won't be caught by CI.

Reply:
Fixed!